src/dreammaker/objtree.rs

```diff
@@ -1140,6 +1140,57 @@ impl ObjectTree {
             }
         }
     }
+
+    // ----------------------------------------------------------------------------
+    // Variable analyzer
+
+    /// Checks for var declaration issues
+    pub fn check_var_defs(&self, context: &Context) {
```

Comment:
Why does this need to be here?

Reply:
it's an examination of the object tree, feels like it belongs here more than just being a global function

Reply:
"checking var defs" is a dreamchecker responsibility so it belongs in dreamchecker IMO. Rust encourages organization by modules rather than just by "most important input type".

Reply:
fair enough